apps/snitch_core/test/data/model/shipping_method_test.exs

```diff
@@ -65,4 +70,79 @@ defmodule Snitch.Data.Model.ShippingMethodTest do
       assert [] = updated_sm.zones
     end
   end
+
+  describe "get/1" do
+    @tag shipping_method_count: 1,
+         state_zone_count: 1,
+         country_zone_count: 1,
+         shipping_category_count: 1
+    test "returns the required shipping method with id", %{shipping_methods: [sm]} do
+      {:ok, new_sm} = ShippingMethod.get(sm.id)
+      assert new_sm.id == sm.id
+    end
+
+    @tag shipping_method_count: 1,
+         state_zone_count: 1,
+         country_zone_count: 1,
+         shipping_category_count: 1
+    test "returns the required shipping method with a map", %{shipping_methods: [sm]} do
+      map = %{name: sm.name}
+      {:ok, new_sm} = ShippingMethod.get(map)
+      assert new_sm.id == sm.id
+    end
+
+    test "fails for invalid id" do
+      assert {:error, :shipping_method_not_found} = ShippingMethod.get(-1)
+    end
+  end
+
+  @tag shipping_method_count: 1,
+       state_zone_count: 1,
+       country_zone_count: 1,
+       shipping_category_count: 1
+  test "get_all/0 returns all the shipping methods succesfully", %{shipping_methods: [sm]} do
+    sm_list = ShippingMethod.get_all()
+    assert length(sm_list) == length([sm])
+  end
+
+  describe "delete/1" do
+    @tag shipping_method_count: 1,
+         state_zone_count: 1,
+         country_zone_count: 1,
+         shipping_category_count: 1
+    test "successfully deletes a shipping method", %{shipping_methods: [sm]} do
+      sm_deleted = ShippingMethod.delete(sm.id)
```

Comment:
This will generate unused `sm_deleted` warning. Also, pattern match on the result returned I think it is `{:ok, _sm}`